syntax = "proto3";  // 指定语法格式，注意 proto3 不再支持 proto2 的 required 和 optional

package user;   // 指定生成的 user.pb.go 的包名，防止命名冲突

// service 定义开放调用的服务，即 NotebookService 微服务
service User {
    // rpc 定义服务内的 GetNotebookInfo 远程调用
    //  方法名  方法参数                 返回值
    /*rpc GetNotebookInfo (NotebookRequest) returns (NotebookResponse);*/
    rpc GetUserInfo (GetUserRequest) returns (GetUserResponse);
}


message GetUserRequest {
    string userId = 1;
}

message MsgUser {
    string userId = 1;     // 用户id
    string updateTime = 2;   // 更新时间
    string createTime = 3;     // 创建时间
    string userName = 4;   // 用户姓名
}


message GetUserResponse {
    // repeated 修饰符表示字段是可变数组，即 slice 类型
    MsgUser data = 1;
}
